### PR TITLE
Opendoor/pr promethus pvc

### PR DESCRIPTION
**Pull request by leochu**
Add persistent volume to Prometheus Chart

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/prometheus/templates/_affinity.tpl [hunk 1]
```diff
@@ -0,0 +1,93 @@
+{{/* affinity - https://kubernetes.io/docs/concepts/configuration/assign-pod-node/ */}}
```
Comment (YenTheFirst) [inline install/kubernetes/helm/istio/charts/prometheus/templates/_affinity.tpl:1]: This commit deserves a detailed commit message about why we're adding `_affinity.tpl` temporarily to this fork, but not adding it to the PR upstream.

File: install/kubernetes/helm/istio/charts/prometheus/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        resources:
+          requests:
+            storage: "{{ .Values.persistentVolume.size }}"
+      {{- if .Values.persistentVolume.storageClass }}
+      {{- if (eq "-" .Values.persistentVolume.storageClass) }}
+        storageClassName: ""
+      {{- else }}
+        storageClassName: "{{ .Values.persistentVolume.storageClass }}"
+      {{- end }}
+      {{- end }}
```
Comment (YenTheFirst) [inline install/kubernetes/helm/istio/charts/prometheus/templates/statefulset.yaml:108]: I prefer
```
{{- if hasKey .Values.persistentVolume "storageClass" }}
  storageClassName: "{{ .Values.persistentVolume.storageClass }}"
{{- end }}
```

I recognize that the current format (falsy, "-", or "name") is popular in the helm chart repo, but there's also at least one example in the wild of this alternative formulation (https://github.com/CenterForOpenScience/helm-charts/blob/master/elasticsearch/templates/data-statefulset.yaml#L190-L192), and I haven't yet seen any deliberate justification for the superiority of (falsy, "-", "name") vs (unset, "", "name") as the set of differing inputs.

File: install/kubernetes/helm/istio/charts/prometheus/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      tolerations:
+{{ toYaml .Values.tolerations | indent 6 }}
+      {{- else if .Values.global.defaultTolerations }}
+      tolerations:
+{{ toYaml .Values.global.defaultTolerations | indent 6 }}
+      {{- end }}
+  volumeClaimTemplates:
+    - metadata:
+        name: storage-volume
+        {{- if .Values.persistentVolume.annotations }}
```
Comment (YenTheFirst) [inline install/kubernetes/helm/istio/charts/prometheus/templates/statefulset.yaml:92]: for these kinds of "if key exists, insert it under a given subkey", I prefer `with` to `if`. i.e.

```
{{- with .Values.persistentVolume.annotations }}
  annotations:
{{ toYaml . | indent 10 }}
{{- end }}

**Review by YenTheFirst — APPROVED** [review]
LGTM, some style things I'd do differently, but those aren't blockers.

File: install/kubernetes/helm/istio/charts/prometheus/templates/statefulset.yaml (hunk 2 above)
Comment (leochu) [reply install/kubernetes/helm/istio/charts/prometheus/templates/statefulset.yaml:108]: This is almost identical except in the case where a user defined the key with no value.

`storageClassName:`

File: install/kubernetes/helm/istio/charts/prometheus/templates/_affinity.tpl (hunk 1 above)
Comment (leochu) [reply install/kubernetes/helm/istio/charts/prometheus/templates/_affinity.tpl:1]: done

**Pull request merged**